### ISSUE TITLE
Can add a layer twice to the QGIS model

### ISSUE THREAD

**Issue by deltamarnix** [Deltares/Ribasim#1683]
- Open QGIS and ensure that the Ribasim plugin is installed and enabled.
- Open the application via the Ribasim button on the QGIS toolbar: _Ribasim panel opens_.
- Press the "New" button in the Model tab: _file navigation window pops up_.
- Fill in a name ("test1").
- Press OK: _A toml and database file are created on the given location_.
- Press the "Edge" layer in the Model tab
- Press "Add to QGIS": _A second Edge layer is added to the model_.

Expected: the "Add to QGIS" button is disabled, because Edge is already in the model.

**Comment by deltamarnix** [comment]
It would perhaps be easier if we don't have this whole synchronization panel in the Model tab. The Layers panel should already tell enough should be the ultimate truth. "Add to QGIS" and "Remove from Dataset" should be removed and the Layers panel should be the geo package itself, no clone.

**Comment by Huite** [comment]
> It would perhaps be easier if we don't have this whole synchronization panel in the Model tab.

I agree, and the synchronization has caused me a fair bit of pain in the past.

> The Layers panel should already tell enough should be the ultimate truth.

This is not the case, and it can't be; the layers panel is not a clone. The plugin is more like a clone of the Browser Panel:

![image](https://github.com/user-attachments/assets/3bb1c4bf-6a0c-42c0-b708-19a8e94615de)

Presence in the layers panel makes QGIS aware of the layer. You can add or remove features, change the styling, etc., but you cannot delete for example. Deleting Layers requires access through the browser panel.
Deleting a layer in the Layers panel simply makes QGIS unaware of the layer's existence.

If you accidentally remove a layer from the Layers panel, it's still in the geopackage and it's still in the model; the data is not deleted. You can only get the layer back by going through the Browser panel and re-adding it to the Layers panel. Alternatively, you just re-open the TOML file and let the plugin do the work.

I think it's worth considering to get rid of the "Add to QGIS" button, and introduce a different button called "Restore (from project)" -- I'll open an issue about that.
But the "Remove from Dataset" button should not be removed, because it's very tedious to delete geopackage layers using the Browser panel.
